filebeat/input/filestream/internal/input-logfile/store.go

```diff
@@ -294,6 +303,29 @@ func (s *store) writeState(r *resource) {
 	}
 }
 
+// resetCursor sets the cursor to the value in cur in the persistent store and
+// drops all pending cursor operations.
+func (s *store) resetCursor(key string, cur interface{}) error {
```

Comment:
All in all the logic looks fine. Still this reset operation and ignoring pending updates is somewhat tricky. Maybe I missed that, but we should have some good unit tests (store_test.go) in order to verify behavior, but also prevent hard to debug regressions in the future.